packages/babel-parser/src/parser/expression.js

```diff
@@ -1153,20 +1156,27 @@ export default class ExpressionParser extends LValParser {
         if (this.state.inPipeline) {
           node = this.startNode();
 
-          if (
-            this.getPluginOption("pipelineOperator", "proposal") !== "smart"
-          ) {
-            this.raise(node.start, Errors.PrimaryTopicRequiresSmartPipeline);
-          }
+          const proposal = this.getPluginOption("pipelineOperator", "proposal");
+          const proposalToNodeType = {
+            hack: "TopicReference",
+            smart: "PipelinePrimaryTopicReference",
+          };
+          const throwPipeTopicRequiresHackPipesError = () => {
+            throw this.raise(node.start, Errors.PipeTopicRequiresHackPipes);
```

Comment:
This is easily recoverable, but +1 for throwing since we cannot choose between `TopicReference` and `PipelinePrimaryTopicReference`.